### PR TITLE
Add UViM project (+misc. changes)

### PR DESCRIPTION
**Pull request by akolesnikoff**
Adds UViM project to big_vision.

### CODE REVIEW

File: big_vision/configs/vit_i21k.py [hunk 1]
```diff
@@ -112,8 +115,26 @@ def get_config(arg=None):
   config.evals.test = {**eval_common, 'split': 'full[:25_600]'}
   config.evals.val = {**eval_common, 'split': 'full[25_600:51_200]'}
   config.evals.train = {**eval_common, 'split': 'full[51_200:76_800]'}
+
+  # Evaluations on ImageNet1k variants
+  eval_common.update(dict(
+      type='classification_with_labelmap', pp_fn=pp_eval_i1k, data_dir=None))
```
Comment (lucasb-eyer) [inline big_vision/configs/vit_i21k.py:121]: You included the configs that use the new `classification_with_labelmap` evaluator, but you did not include the evaluator itself. Please either also include it (and mention that in the PR description), or exclude these changes for now. (same in the bit_i21k.py)

File: big_vision/requirements.txt [hunk 2]
```diff
@@ -4,7 +4,9 @@ einops
 flax
 git+https://github.com/google/flaxformer
 git+https://github.com/deepmind/optax.git  # Needed for decay_rate_fn in scale_by_factored_rms
+git+https://github.com/akolesnikoff/panopticapi.git@mute
```
Comment (lucasb-eyer) [inline big_vision/requirements.txt:7]: Nice :)

File: big_vision/utils.py [hunk 3]
```diff
@@ -685,44 +755,84 @@ def recover_tree(keys, values):
...
+  if default is ValueError:
+    raise ValueError(
+        f"Cannot convert {prefix} to steps, due to missing batch_size "
+        f"({batch_size}), data_size ({data_size}), or corresponding entry in "
+        "config:\n" + "\n".join(config.keys()))
+
+  return default
+
+
 def create_learning_rate_schedule(
```
Comment (lucasb-eyer) [inline big_vision/utils.py:808]: fPushing this update means the distillation and the image_text trainers will not work anymore, and even your uvim trainers should not actually be working with this.
See call to `bv_optax.make` in all of them vs in train.py. global_batch_size -> batch_size, and also -steps_per_epoch but +data_size.

Ideally you could change these accordingly, I could do that too, but I'm really short on time until Monday evening.

**Review by lucasb-eyer — CHANGES_REQUESTED** [review]

File: big_vision/utils.py (hunk 3 above)
Comment (andresusanopinto) [reply big_vision/utils.py:808]: @lucasb-eyer - I noticed the same thing, but then looked in details I realised that the current state is messy and error-prone but not broken at head. I suggest we fix it after this and not try to roll the fix with this one. Alternative lets fix it and do a pull-request before uvim is added.

In detail, note:
 - The old global_batch_size was only need in utils.create_learning_rate_schedule when scale_with_batchsize is set to True. This never happens in any submitted config. This is now replaced by a use of batch_size.. but again those lines don't trigger in any subbmitted config.
 - The old "steps_per_epoch" was used to compute "warmup/cooldown_epochs" which was not used by any config. It is now replaced by a function which depends on the new batch_size and data_size.. but again those are only needed when one has a config with "warmup/cooldown_{epochs/examples}" which is also not used in any config.

Comment (lucasb-eyer) [reply big_vision/utils.py:808]: Yep, your analysis is exactly correct. OK either way for me to fix this after, or before. It's really a simple change, I might quickly make it internally so you can squeeze it in here, but no need to wait for it.

File: .gitignore [hunk 4]
```diff
@@ -1 +0,0 @@
-__pycache__
```
[the remark this answers is not in the document]
Comment (akolesnikoff) [reply .gitignore]: returned this file. It was not properly added in the first place, but now all should be fine.

File: big_vision/configs/vit_i21k.py (hunk 1 above)
Comment (akolesnikoff) [reply big_vision/configs/vit_i21k.py]: excluded from opensourcing.
